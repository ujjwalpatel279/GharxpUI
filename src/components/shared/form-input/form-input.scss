// ========================================================================
// COMPONENTS/SHARED/FORM-INPUT
// ========================================================================

@import '~sassHelpers';

.c-form-input {
  height: spacer(10);
  padding-left: spacer(2);
  padding-right: spacer(2);
  border: 1px solid color(grey2);
  border-radius: $border-radius;
  background-color: white;

  &.isInvalid {
    background-color: color(errorLight);
    border-color: color(errorDark);
  }

  &:hover,
  &:focus {
    border-color: var(--primary);
  }
}
