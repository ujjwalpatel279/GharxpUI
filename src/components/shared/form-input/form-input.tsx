import React, { forwardRef } from 'react';

interface FormFieldProps {
  id: string;
  type?: 'text' | 'email' | 'number' | 'tel';
  value?: string;
  placeHolder?: string;
  error?: boolean;
}

import './form-input.scss';

export const FormInput = forwardRef(
  ({ id, type = 'text', value, placeHolder, error }: FormFieldProps, ref: React.Ref<HTMLInputElement>) => {
    let inputClasses = 'c-form-input';

    if (error) {
      inputClasses += ' isInvalid';
    }
    return (
      <input
        type={type}
        id={id}
        name={id}
        defaultValue={value}
        placeholder={placeHolder}
        className={inputClasses}
        ref={ref}
        aria-invalid={error ? 'true' : 'false'}
      />
    );
  },
);

FormInput.displayName = 'FormInput';
