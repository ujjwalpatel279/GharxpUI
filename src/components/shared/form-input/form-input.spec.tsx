import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';

import { FormInput } from './form-input';

describe('FormInput Component', () => {
  afterEach(cleanup);

  it('renders the form input component', () => {
    const { container } = render(<FormInput id="firstName" ref={() => null} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('aria-invalid is false', () => {
    const { queryByPlaceholderText } = render(
      <FormInput id="firstName" error={false} placeHolder="testInput" ref={() => null} />,
    );

    expect(queryByPlaceholderText('testInput').getAttribute('aria-invalid')).toBe('false');
  });

  it('aria-invalid is true', () => {
    const { queryByPlaceholderText } = render(
      <FormInput id="firstName" error={true} placeHolder="testInput" ref={() => null} />,
    );

    expect(queryByPlaceholderText('testInput').getAttribute('aria-invalid')).toBe('true');
  });

  it('input value is initially empty and has a value once a value is entered', () => {
    const { queryByPlaceholderText } = render(
      <FormInput id="firstName" error={true} placeHolder="testInput" ref={() => null} />,
    );

    const input = queryByPlaceholderText('testInput') as HTMLInputElement;
    expect(input.value).toBe('');

    fireEvent.change(input, { target: { value: 'Rafael' } });
    expect(input.value).toBe('Rafael');
  });
});
