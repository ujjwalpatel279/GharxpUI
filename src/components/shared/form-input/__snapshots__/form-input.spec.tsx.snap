// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput Component renders the form input component 1`] = `
<input
  aria-invalid="false"
  class="c-form-input"
  id="firstName"
  name="firstName"
  type="text"
  value=""
/>
`;
